crates/matrix-sdk-common/src/js_tracing.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{fmt, fmt::Write};
+
+use tracing::{
+    field::{Field, Visit},
+    Event, Level, Subscriber,
+};
+use tracing_subscriber::layer::Context;
+
+//! Utilities for `tracing` in JS environments
```

Comment:
oh man. Clippy complained when this was `///!`, and now rustfmt is complaining about `//!`. Isn't there a way to document a module like this?

Reply:
>  note: inner doc comments like this (starting with `//!` or `/*!`) can only appear before items

I thought the whole *point* of an inner doc comment was that it could appear *inside* the thing it was documented, rather than *before* it.

Reply:
It's trying to tell you that the module documentation has to come before the imports.

Reply:
(the "items" being the `use` statements)